File: daily-programmer/1-easy-input/cpp/main.cpp
#include "utils.hpp"

using namespace std;
using namespace utils;

int main()
{
    string name = prompt("Input name > ");
    string age = prompt("Input age > ");
    string username = prompt("Input username > ");

    cout << "your name is (" << name
         << "), you are (" << age
         << ") years old, and your username is (" << username << ")" << endl;

    ofstream os("log.txt", ios_base::app);
    if (!os.is_open()) {
        cerr << "Failed to open 'log.txt' file";
        exit(EXIT_FAILURE);
    }

    os << "name: " << name
       << ", age: " << age
       << ", username: " << username << endl;
}

File: daily-programmer/1-easy-input/rust/src/main.rs
use std::io;

fn main() {

    print!("Input name > ");
    let mut name = String::new();
    io::stdin().read_line(&mut name).unwrap();

    print!("Input years > ");
    let mut years = String::new();
    io::stdin().read_line(&mut years).unwrap();

    print!("Input username > ");
    let mut username = String::new();
    io::stdin().read_line(&mut username).unwrap();

    println!("Your name is {0}, you are {1} years old, your username is {2}", name, years, username);
	
}
